helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/CodeTransformations.java

```diff
@@ -548,6 +553,38 @@ public static String removeAvroGeneratedAnnotation(String code, AvroVersion minS
     return AVROGENERATED_ANNOTATION_PATTERN.matcher(code).replaceAll("");
   }
 
+  /**
+   * MODEL$ was introduced as a static field in the generated class, starting 1.8, which indicates the conversions
+   * applicable for a specific record. However, starting 1.9, avro uses reflection to look up this field, which will throw
+   * a {@link ReflectiveOperationException exception for records generated from older version. This results in  performance
+   * degradation.
+   * This methods avoids the exception by introducing this field in older versions of the generated record.
+   *
+   * @param code avro generated code that may or may not have the MODEL$ declaration
+   * @param minSupportedVersion lowest avro version under which the generated code should work
+   * @param maxSupportedVersion highest avro version under which the generated code should work
+   * @return code where MODEL$ exists for avro versions expecting it at runtime (>= 1.9)
+   */
+  public static String pacifyModel$Delcaration(String code, AvroVersion minSupportedVersion, AvroVersion maxSupportedVersion) {
+    if (minSupportedVersion.laterThan(AvroVersion.AVRO_1_7)) { // min >= 1.8
+      return code;
+    }
+    if (maxSupportedVersion.earlierThan(AvroVersion.AVRO_1_8)) { // max < 1.8
+      // remove any model annotation as it will not be used
+      return MODEL_DECL_PATTERN.matcher(code).replaceAll("");
+    } else {
+      // If MODEL$ is already present in the code, replace it with SpecificData#get() instead of SpecificData ctor, which was
+      // made public only from avro 1.7
+      if (MODEL_DECL_PATTERN.matcher(code).find()) {
+        return MODEL_DECL_PATTERN.matcher(code).replaceAll(Matcher.quoteReplacement(MODEL_DECL_REPLACEMENT));
+      } else {
+        // add it to the end of generated code to avoid breaking
+        int schema$EndPosition = findEndOfSchemaDeclaration(code);
```

Comment:
is it possible to insert it roughtly where it would be if generated under modern-ish vanilla avro? i understand appending at the end is easier to do, but remaining close to vanilla code has advantages - for example all the lookup patterns in this class assume vanilla code struucture

Reply:
ok. let me check where it inserts in the modern vanilla avro and try to add it there.

Reply:
I verified that modern vanilla avro is adding `MODEL$` declaration right after the `SCHEMA$` declaration.
Example, in avro 1.10, I see the following generated code:

```
package by110;
...
@AvroGenerated
public class SimpleRecord extends SpecificRecordBase implements SpecificRecord {
  private static final long serialVersionUID = -6820601432685633308L;
  public static final Schema SCHEMA$ = (new Parser()).parse("{\"type\":\"record\",\"name\":\"SimpleRecord\",\"namespace\":\"by110\",\"fields\":[{\"name\":\"stringField\",\"type\":\"string\"}]}");
  private static SpecificData MODEL$ = new SpecificData();
  private static final BinaryMessageEncoder<SimpleRecord> ENCODER;
```